tests/ui/assign_ops2.rs

```diff
@@ -53,3 +53,17 @@ impl MulAssign<i64> for Wrap {
         *self = *self * rhs
     }
 }
+
+fn cow_add_assign() {
+    use std::borrow::Cow;
+    let mut buf = Cow::Owned(String::from("bar"));
+    let cows = Cow::Borrowed("foo");
+
+    // this can be linted
+    buf = buf + cows.clone();
+
+    // this should not as cow<str> Add is not commutative
+    buf = cows + buf;
+    println!("{}", buf);
+
+}
```

Comment:
missing newline at end of file

Reply:
I admit I don't usually pay attention to it… And I'm surprised you managed to see it via Github UI.
I don't seem to see it in the diffs.

Anyways! I added it.